src/Mcfedr/QueueManagerBundle/Command/PubSubRunnerCommand.php

```diff
@@ -0,0 +1,172 @@
+            $jobs = $this->getJobsFromSubscription($subscription);
+            if ($jobs) {
+                return $jobs;
+            }
+        }
+
+        return null;
+    }
+
+    protected function finishJobs(JobBatch $batch): void
+    {
+        $retryJobs = $batch->getRetries();
+        //todo implement retry failed jobs
+        /** @var PubSubJob[] $toDelete */
+        $toDelete = array_merge($batch->getOks(), $batch->getRetries(), $batch->getFails());
+        if (\count($toDelete)) {
+            $subscription = $toDelete[0]->getSubscription();
+            $toDelete = array_map(function (PubSubJob $message) {
+                return $pubSubMessage = (new Message(['messageId' => $message->getId()], ['ackId' => $message->getAckId()]));
+            }, $toDelete);
+
+            $this->pubSub->subscription($subscription)->acknowledgeBatch($toDelete);
+        }
+    }
+
+    protected function handleInput(InputInterface $input): void
+    {
+        if (($url = $input->getOption('url'))) {
+            $this->subscriptions = explode(',', $url);
+        } elseif (($queue = $input->getOption('queue'))) {
+            $this->subscriptions = array_map(function ($queue) {
+                return $this->subscriptions[$queue];
+            }, explode(',', $queue));
+        } else {
+            $this->subscriptions = [$this->defaultSubscription];
+        }
+
+        if (($batch = $input->getOption('batch-size'))) {
+            $this->batchSize = (int) $batch;
+        }
+
+        if (($waitTime = $input->getOption('wait-time'))) {
+            $this->waitTime = (int) $waitTime;
+        }
+    }
+
+    private function setOptions(array $options): void
+    {
+        $this->defaultSubscription = $options['default_subscription'];
+        $this->subscriptions = array_map(function ($topic) {
+            return $topic['subscribtion'];
+        }, $options['topics']);
+
+        if (!\array_key_exists('default', $this->subscriptions)) {
+            $this->subscriptions['default'] = $this->defaultSubscription;
+        }
+    }
+
+    private function getJobsFromSubscription($subscription): ?JobBatch
+    {
+        $response = $this->pubSub->subscription($subscription)->pull(['maxMessages' => $this->batchSize]);
+
+        if (\count($response)) {
+            $jobs = [];
+            $exception = null;
+
+            /** @var Message $message */
+            foreach ($response as $message) {
+                if ($message->data() === null || (!\is_array($data = json_decode($message->data(), true))) || !isset($data['name']) || !isset($data['arguments']) || !isset($data['retryCount'])) {
+                    $exception = new UnexpectedJobDataException('Sqs message(s) missing data fields name, arguments and retryCount');
+                    $pubSubMessage = (new Message(['messageId' => $message->id()], ['ackId' => $message->ackId()]));
+                    $this->pubSub->subscription($subscription)->acknowledge($pubSubMessage);
```

Comment:
Would be good to change this to acknowledge in a single batch request, similar to how its done in the `SqsRunner` - i.e. you can make an array and make the batch call after.